kernel/kernel-api/src/main/java/io/delta/kernel/Transaction.java

```diff
@@ -142,6 +194,16 @@ static CloseableIterator<Row> generateAppendActions(
             Row transactionState,
             CloseableIterator<DataFileStatus> fileStatusIter,
             DataWriteContext dataWriteContext) {
-        throw new UnsupportedOperationException("Not implemented yet");
+        URI tableRoot = new Path(getTableRoot(transactionState)).toUri();
+        return fileStatusIter.map(
+                dataFileStatus -> {
+                    Row addFileRow = AddFile.convertDataFileStatus(
+                            tableRoot,
+                            dataFileStatus,
+                            dataWriteContext.getPartitionValues(),
```

Comment:
Aren't these now lower case keys? Isn't that incorrect or is that how they are stored?